notebooks/ima_viewer/ima_tools.py

```diff
@@ -0,0 +1,747 @@
+                            options=ext_options,
+                            description='Extension: ',
+                            layout=Layout(width='50%'))
+    lim_slider=widgets.FloatRangeSlider(min=0, max=100,
+                                        value=lim_value,
+                                        step=lim_step,
+                                        readout_format='.1f',
+                                        description='Scaling (%): ',
+                                        layout=Layout(width='50%'))
+    return cmap, ext, lim_slider
+
+
+def verify_input(filepath, ext_type):
+    """
+    Verify file exists and the extension is valid.
+
+    Parameters
+    ----------
+    filepath : str
+        String representation of the path to the
+        IMA file.
+    ext_type : str
+        Extension of the IMA fits file that should
+        be displayed. Valid inputs are 'SCI', 'ERR',
+        and 'DQ'. Default is 'SCI'.
+
+    Returns
+    -------
+    inputs : tuple of str or None
+        If both the filepath and the extension are
+        valid, returns a tuple of format (filepath,
+        ext). Otherwise, returns None.
+    """
+    if os.path.exists(filepath):
+        valid_ext = verify_ext(ext_type)
+        if valid_ext != None:
+            inputs = (filepath, valid_ext)
+        else:
+            print('Filepath verified, but extension was not valid.')
+            inputs = None
+    else:
+        print('Filepath could not be verified.')
+        inputs = None
+
+    return inputs
+
+
+def verify_ext(ext_type):
+    """
+    Parameters
+    ----------
+    ext_type : str
+        Extension of the IMA fits file that should
+        be displayed. Valid inputs are 'SCI', 'ERR',
+        and 'DQ'. Default is 'SCI'.
+
+    Returns
+    -------
+    valid_ext : str or None
+        If entered extension name (or uppercase version)
+        is a valid extension, returns the extension name.
+        Otherwise, returns None.
+    """
+    ext_type = ext_type.upper()
+    if ext_type in ['SCI', 'DQ', 'ERR']:
+        valid_ext = ext_type
+
+    else:
+        if ext_type in ['PRIMARY', 'SAMP', 'TIME']:
+            message = f'Cannot display data as image for extension type: {ext_type}'
+        else:
+            message = f'{ext_type} is not a recognized extension for IMA files.'
+        print(f'{message}\nPossible extensions for image data are:\n\tSCI\n\tDQ\n\tERR')
+        valid_ext = None
+
+    return valid_ext
+
+
+# output functions
+def get_ax_info(ax):
+    """
+    Grabs the rootname and the x- and y-limits
+    from the current ax view
+
+    Parameters
+    ----------
+    ax : matplotlib.axes._subplots.AxesSubplot
+
+    Returns
+    -------
+    rootname : str
+        Rootname identifier for the current
+        observation.
+
+    xlims, ylims : tuples of int
+
+    """
+    rootname = ax.get_title().split(':')[0]
+
+    x1, x2 = ax.get_xlim()
+    y1, y2 = ax.get_ylim()
+
```

Comment:
Functionality for 2.0:

Make xlim and ylim parameters for functions. Then users can make images/gifs zoomed in and not just full frames.